theLudovyc/GUI/ResourceControl.gd

```diff
@@ -4,11 +4,13 @@ extends PanelContainer
 
 @onready var label_amount := $HBoxContainer/PanelContainer/HBoxContainer/LabelAmount
 @onready var label_production_rate := $HBoxContainer/PanelContainer/HBoxContainer/LabelProductionRate
+@onready var texture_button: ResourceButton = $TextureButton
 
 
 # Called when the node enters the scene tree for the first time.
 func _ready():
-	$TextureButton.set_resource_icon(resource_type)
+	texture_button.set_resource_icon(resource_type)
```

Comment:
If the variable is only used one time, or only in the _ready, instead of creating @onready var who took memory, you can use $